sequencer/src/bin/deploy.rs

```diff
@@ -104,95 +85,26 @@ struct Options {
-    let wallet = MnemonicBuilder::<English>::default()
-        .phrase(opt.mnemonic.as_str())
-        .index(opt.account_index)?
-        .build()?
-        .with_chain_id(chain_id);
-    let owner = wallet.address();
-    let l1 = Arc::new(SignerMiddleware::new(provider, wallet));
-
-    // As a sanity check, check that the deployer address has some balance of ETH it can use to pay
-    // gas.
-    let balance = l1.get_balance(owner, None).await?;
-    ensure!(
-        balance > 0.into(),
-        "deployer account {owner:#x} is not funded!"
-    );
-    tracing::info!(%balance, "deploying from address {owner:#x}");
-
-    // `HotShot.sol`
-    if should_deploy(ContractGroup::HotShot, &opt.only) {
-        contracts
-            .deploy_tx(Contract::HotShot, HotShot::deploy(l1.clone(), ())?)
-            .await?;
-    }
-
-    // `LightClient.sol`
-    if should_deploy(ContractGroup::LightClient, &opt.only) {
-        // Deploy the upgradable light client contract first, then initialize it through a proxy contract
-        let lc_address = if opt.use_mock_contract {
-            contracts
-                .deploy_fn(Contract::LightClient, |contracts| {
-                    deploy_mock_light_client_contract(l1.clone(), contracts, None).boxed()
-                })
-                .await?
-        } else {
-            contracts
-                .deploy_fn(Contract::LightClient, |contracts| {
-                    deploy_light_client_contract(l1.clone(), contracts).boxed()
-                })
-                .await?
-        };
-        let light_client = LightClient::new(lc_address, l1.clone());
-
-        let genesis = light_client_genesis(&opt.orchestrator_url, opt.stake_table_capacity).await?;
-        let data = light_client
-            .initialize(genesis.into(), u32::MAX, owner)
-            .calldata()
-            .context("calldata for initialize transaction not available")?;
-        contracts
-            .deploy_tx(
-                Contract::LightClientProxy,
-                ERC1967Proxy::deploy(l1.clone(), (lc_address, data))?,
-            )
-            .await?;
-    }
-
-    // `FeeContract.sol`
-    if should_deploy(ContractGroup::FeeContract, &opt.only) {
-        let fee_contract_address = contracts
-            .deploy_tx(Contract::FeeContract, FeeContract::deploy(l1.clone(), ())?)
-            .await?;
-        let fee_contract = FeeContract::new(fee_contract_address, l1.clone());
-        let data = fee_contract
-            .initialize(owner)
-            .calldata()
-            .context("calldata for initialize transaction not available")?;
-        contracts
-            .deploy_tx(
-                Contract::FeeContractProxy,
-                ERC1967Proxy::deploy(l1.clone(), (fee_contract_address, data))?,
-            )
-            .await?;
-    }
+    let contracts = Contracts::from(opt.contracts);
+
+    let genesis = light_client_genesis(&opt.orchestrator_url, opt.stake_table_capacity).await?;
```

Comment:
native demo is failing because this binary is waiting for pubkeys from orchestrator
`Peers' keys are not ready, retrying.`

Reply:
@ImJeremyHe @imabdulbasit I think the deployment of the sequencer-contracts (fee contract) must work without needing the orchestrator. I think that's why the deployment is split into two tasks (sequencer / prover contracts).

Currently in this PR the orchestrator needs the sequencers to provide the peer configs and the deploy-sequencer-contracts needs to orchestrator and the sequencers need the deploy-sequencer-contract so there's a circular dependency.

Reply:
Changed the `genesis` parameter to be a `BoxFuture` that is awaited only for `LightClient` contract.  We can also split it into separate functions for each contract.

Reply:
76dec3068f2d29c1a582a5693a3a1eae4e26cc06